This create option BUILD_TESTS for miopen, like other rocm math libs, and
testing executables are always built if test enabled. The original behaviour, is
always add test/CMakeLists.txt without adding test exes as targets, and compile them
during test.
Index: MIOpen/CMakeLists.txt
===================================================================
--- MIOpen.orig/CMakeLists.txt
+++ MIOpen/CMakeLists.txt
@@ -748,8 +748,11 @@ add_subdirectory(src)
 if(MIOPEN_BUILD_DRIVER)
     add_subdirectory(driver)
 endif()
-add_subdirectory(test)
-add_subdirectory(speedtests)
+option(BUILD_TESTS "Build binaries for tests" OFF)
+if(BUILD_TESTS)
+	add_subdirectory(test)
+	add_subdirectory(speedtests)
+endif()
 add_subdirectory(utils)
 if(MIOPEN_ENABLE_FIN)
 add_subdirectory(fin)
Index: MIOpen/test/CMakeLists.txt
===================================================================
--- MIOpen.orig/test/CMakeLists.txt
+++ MIOpen/test/CMakeLists.txt
@@ -301,7 +301,7 @@ endfunction()
 separate_arguments(MIOPEN_TEST_FLAGS_ARGS UNIX_COMMAND ${MIOPEN_TEST_FLAGS})
 
 function(add_test_executable TEST_NAME)
-    add_executable (${TEST_NAME} EXCLUDE_FROM_ALL ${ARGN})
+    add_executable (${TEST_NAME} ${ARGN})
     clang_tidy_check(${TEST_NAME})
     target_link_libraries(${TEST_NAME} ${CMAKE_THREAD_LIBS_INIT})
     # Cmake does not add flags correctly for gcc
@@ -1910,4 +1910,4 @@ add_custom_test(test_regression_half_mi2
 )
 #override if we need to install gtests
 set(INSTALL_GTEST OFF)
-add_subdirectory(gtest EXCLUDE_FROM_ALL)
+add_subdirectory(gtest)
Index: MIOpen/speedtests/CMakeLists.txt
===================================================================
--- MIOpen.orig/speedtests/CMakeLists.txt
+++ MIOpen/speedtests/CMakeLists.txt
@@ -7,7 +7,7 @@ else()
 endif()
 
 function(add_speedtest_executable TEST_NAME)
-    add_executable (${TEST_NAME} EXCLUDE_FROM_ALL ${ARGN})
+    add_executable (${TEST_NAME} ${ARGN})
     clang_tidy_check(${TEST_NAME})
     target_link_libraries(${TEST_NAME} ${CMAKE_THREAD_LIBS_INIT})
     # Cmake does not add flags correctly for gcc
